src/tiling.rs

```diff
@@ -863,8 +863,8 @@ impl Primitive {
                                                        details.tl_inner.x,
                                                        details.tl_inner.y),
                     },
-                    vertical_color: details.top_color,
```

Comment:
This will break inset borders. Let's just revert the colors back and keep the top left as is. Sorry for reverting the change. The previous commit works fine. Thanks again!